modules/samples/extensions/geo-event-fuser/src/main/java/org/wso2/siddhi/extension/geo/geoeventfuser/FuseEvents.java

```diff
@@ -29,66 +26,68 @@
 import org.wso2.siddhi.core.executor.VariableExpressionExecutor;
 import org.wso2.siddhi.core.query.processor.Processor;
 import org.wso2.siddhi.core.query.processor.stream.window.WindowProcessor;
+import org.wso2.siddhi.query.api.definition.Attribute;
 import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
 
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.HashMap;
 import java.util.List;
-import org.apache.commons.logging.Log;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class FuseEvents extends WindowProcessor {
-
-    private String variable = "";
-    private int variablePosition = 0;
-    private HashMap<String, ArrayList<StreamEvent>> eventsBuffer = null;
-
-    private static final Log log = LogFactory.getLog(FuseEvents.class);
-
-    @Override
+    private final String[] statesArray = new String[]{"OFFLINE", "NORMAL", "WARNING", "ALERTED"};
```

Comment:
@sajithshn I agree. However, the `fuseEvent(StreamEvent event)` method requires both the string value and the index value of the given event's state. That's the reason to use a simple array instead of an enum.